app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -293,58 +252,61 @@ class ThreadFragment : Fragment() {
     }
 
     private fun setupAdapter(result: OpenThreadResult) = with(binding.messagesList) {
-
         addItemDecoration(DividerItemDecorator(InsetDrawable(dividerDrawable(context), 0)))
         recycledViewPool.setMaxRecycledViews(0, 0)
+        threadAdapter.stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
+        updateAdapter(result)
+    }
 
-        threadAdapter.apply {
-            isExpandedMap = result.isExpandedMap
-            initialSetOfExpandedMessagesUids = result.initialSetOfExpandedMessagesUids
-            isThemeTheSameMap = result.isThemeTheSameMap
-
-            stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
-            contacts = mainViewModel.mergedContactsLive.value ?: emptyMap()
-        }
+    private fun setupDialogs() {
+        bindAlertToViewLifecycle(descriptionDialog)
+        linkContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
+        emailContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
+        phoneContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
     }
 
-    private fun observeThreadLive() {
-        threadViewModel.threadLive.observe(viewLifecycleOwner, ::onThreadUpdate)
+    private fun observeLightThemeToggle() {
+        mainViewModel.toggleLightThemeForMessage.observe(viewLifecycleOwner, threadAdapter::toggleLightMode)
     }
 
-    private fun onThreadUpdate(thread: Thread?) = with(binding) {
+    private fun observeThreadLive() = with(binding) {
 
-        if (thread == null) {
-            leaveThread()
-            return@with
-        }
+        threadViewModel.threadLive.observe(viewLifecycleOwner) { thread ->
 
-        threadSubject.movementMethod = LinkMovementMethod.getInstance()
+            if (thread == null) {
+                closeThread()
+                return@observe
+            }
 
-        iconFavorite.apply {
-            setIconResource(if (thread.isFavorite) R.drawable.ic_star_filled else R.drawable.ic_star)
-            val color = if (thread.isFavorite) {
-                context.getColor(R.color.favoriteYellow)
-            } else {
-                context.getAttributeColor(RMaterial.attr.colorPrimary)
+            threadSubject.movementMethod = LinkMovementMethod.getInstance()
+
+            iconFavorite.apply {
+                setIconResource(if (thread.isFavorite) R.drawable.ic_star_filled else R.drawable.ic_star)
+                val color = if (thread.isFavorite) {
+                    context.getColor(R.color.favoriteYellow)
+                } else {
+                    context.getAttributeColor(RMaterial.attr.colorPrimary)
+                }
+                iconTint = ColorStateList.valueOf(color)
```

Comment:
This logic shouldn't be in the fragment but in the `ViewModel` instead. Instead of returning a simple thread, you can also return the icon and the color and tint to use.

Reply:
This is the way we currently do probably everywhere in the app.
I'm open to suggestions to change it, but we should probably discuss it with the team before, because if we change this, it should be app-wide.